src/core/net/ip6.cpp

```diff
@@ -672,7 +904,12 @@ otError Ip6::HandlePayload(Message &aMessage, MessageInfo &aMessageInfo, uint8_t
     switch (aIpProto)
     {
     case kProtoUdp:
-        ExitNow(error = mUdp.HandleMessage(aMessage, aMessageInfo));
+        error = mUdp.HandleMessage(aMessage, aMessageInfo);
+        if (error == OT_ERROR_DROP)
+        {
+            otLogCritIp6("Error UDP Checksum");
```

Comment:
Not sure I would put this log at the critical level given that the received message itself is generated by anyone externally. What about making this notice level?

Reply:
Make sense.